### PR TITLE
ROX-25450: Remap imagePullErr

### PR DESCRIPTION
**Pull request by lvalerom**
### Description

Make the `ErrImagePull` and `ImagePullBackOff` of the upgrader deployment more readable in the UI. For other reasons, remove the low-level reason message.

I did a little bit of digging in the differences between `ImagePullBackOff`, `ErrImagePull`, and `ErrImageNeverPull`:

* `ErrImagePull`: Is an error pulling the image.
* `ImagePullBackOff`: Is the state of waiting grace period after an `ErrImagePull`.
* `ErrImageNeverPull`: For what I could find this happens if the image was never pulled `imagePullPolicy: Never` (and it's not present in the cluster?).

With this I decided to add remediation for `ErrImagePull` and `ImagePullBackOff` for now but leave the rest out.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- ~[ ] added unit tests~
- ~[ ] added e2e tests~
- ~[ ] added regression tests~
- ~[ ] added compatibility tests~
- ~[ ] modified existing tests~

#### How I validated my change

- [x] Manual testing overwriting the upgrader image with [7ccbb98](https://github.com/stackrox/stackrox/pull/12257/commits/7ccbb9852f7b39adf22bc12801573da0bb2aab30)
See: 

Screenshot 2024-08-02 at 19 46 56

### CODE REVIEW

File: sensor/kubernetes/upgrade/process.go [hunk 1]
```diff
@@ -400,7 +400,7 @@ func (p *process) checkPodStatus(pod *v1.Pod) *central.UpgradeCheckInFromSensorR
 	} else if waitingState := upgraderContainerStatus.State.Waiting; waitingState != nil {
 		if isImagePullRelatedReason(waitingState.Reason) {
 			s.Error = &central.UpgradeCheckInFromSensorRequest_PodErrorCondition{
-				Message:      fmt.Sprintf("Error pulling image: %s (%s)", waitingState.Reason, waitingState.Message),
+				Message:      fmt.Sprintf("The upgrader had trouble pulling the new \"%s\" image. This typically happens when the image does not exist or Sensor lacks required credentials to pull it.", upgraderContainerStatus.Image),
 				ImageRelated: true,
```
Comment (rukletsov) [inline sensor/kubernetes/upgrade/process.go:404]: Thank you for adding remediation! I'd suggest to add it only to the reason we know it applies to, `ErrImagePull`, and keep the generic message for the rest. Something like 

```
var string remediation
switch waitingState.Reason {
    case "ErrImagePull", "ImagePullBackOff", "ErrImageNeverPull": remediation = " This typically happens when the image does not exist or Sensor lacks required credentials to pull it."
}

<...>
Message: fmt.Sprintf("The upgrader had trouble pulling the new \"%s\" image.%s (Reason: %s)", upgraderContainerStatus.Image, remediation, waitingState.Reason)
```

Comment (lvalerom) [reply sensor/kubernetes/upgrade/process.go:404]: Done. I added the function `getImageErrorRemediation` so that in the future is cleaner to extend if we want to add more remediations.

Comment (lvalerom) [reply sensor/kubernetes/upgrade/process.go:404]: I did a little bit of digging in the differences between `ImagePullBackOff`, `ErrImagePull`, and `ErrImageNeverPull`:

- `ErrImagePull`: Is an error pulling the image.
- `ImagePullBackOff`: Is the state of waiting grace period after an `ErrImagePull`.
- `ErrImageNeverPull`: For what I could find this happens if the image was never pulled `imagePullPolicy: Never` (and it's not present in the cluster?).

With this I would definitely add `ErrImagePull` but I think I'll leave `ErrImageNeverPull` out. WDYT?

Comment (rukletsov) [reply sensor/kubernetes/upgrade/process.go:404]: Sounds good, thank you. We can always extend the switch block as we learn about new edge cases.

**Review by rukletsov — APPROVED** [review]
Modulo test code, please don't forget to remove it!
